thetis/exporter.py

```diff
@@ -5,6 +5,23 @@
 """
 from utility import *
 import ufl
+from firedrake.output import is_cg
+
+
+def get_visu_space(fs):
+    """
+    Figure out the appropriate linear visualization space for fs
+    """
+    is_vector = len(fs.ufl_element().value_shape()) == 1
+    mesh = fs.mesh()
+    family = 'Lagrange' if is_cg(fs) else 'Discontinuous Lagrange'
+    if is_vector:
+        visu_fs = VectorFunctionSpace(mesh, family, 1)
+    else:
+        visu_fs = FunctionSpace(mesh, family, 1)
```

Comment:
Most of the function spaces created here already exist in the Thetis solver object. I presume that function spaces are cached somehow, and this does not actually create any new instances?